crates/sc-consensus-subspace/src/block_import.rs

```diff
@@ -672,18 +683,39 @@ where
             }
         }
 
-        // The fork choice rule is that we pick the heaviest chain (i.e. smallest solution range),
-        // if there's a tie we go with the longest chain
+        // The fork choice rule is the largest fork solution range weight.
+        //
+        // This almost always prioritises:
+        // - the longest chain (the largest number of solutions), and if there is a tie
+        // - the strictest solutions (the numerically smallest solution ranges).
+        //
+        // If these totals are equal:
+        // - each node keeps the block it already chose (the one that it processed first).
+        //
+        // If there is no previous best block, or the old best block is missing a weight, or has a
+        // zero weight:
+        // - the new block is chosen as the best block, as long as it has a non-zero weight.
+        // (The only blocks with zero weights are in the test runtime.)
+        //
+        // Solution ranges only change at the end of each era, where different block times can make
+        // the range in each fork different. This can lead to some edge cases:
+        // - one fork accepts a solution as within its range, but another with a narrower range
+        //   does not, or
```

Comment:
I don't think this description makes a lot of sense. Era transition is based on the block number. So if you have different solutions, the parents must be different as well. A different fork will not accept the block not because its solution is not good enough, but rather because the parent is invalid/unknown to begin with.

Reply:
Ok, this would be clearer as:
```suggestion
        // - one fork accepts a solution as within its range, but another with a narrower range
        //   does not accept another solution with similar difficulty, or
```

It was meant to be a security argument about difficulty, not solution re-use.